File: 41.FirstMissingPositive.java
import java.util.HashSet;

public class FirstMissingPositive {
    public static void main(String[] args) {
        int nums[] = { 1, 2, 0 };
        Solution1 sol = new Solution1();
        sol.firstMissingPositive(nums);
    }
}

class Solution1 {
    public int firstMissingPositive(int[] nums) {
        int count = 1;
        HashSet<Integer> numsSet = new HashSet<>();
        for (int num : nums) {
            numsSet.add(num);
        }
        while (true) {
            if (numsSet.contains(count)) {
                count++;
            } else {
                return count;
            }
        }
    }
}

File: 41.FirstMissingPositive.py
class Solution(object):
    def firstMissingPositive(self, nums):
        group = set(nums)
        count = 1
        print(group)
        while True:
            if count in group:
                count +=1
            else:
                return count

# Prueba
nums = [1,2]
sol = Solution()
r = sol.firstMissingPositive(nums)
print(r)  